packages/adapter-mongoose/tokenizers/relationship-path.js

```diff
@@ -29,15 +29,16 @@ module.exports = function getRelatedListAdapterFromQueryPathFactory(listAdapter)
       }
 
       // Then follow the breadcrumbs to find the list adapter
+      const currentKey = foundListAdapter.key;
       foundListAdapter = fieldAdapter.getRefListAdapter();
 
       if (!foundListAdapter) {
```

Comment:
I don't get this, if `foundListAdapter` is falsy then trying to get `foundListAdapter.key` will throw so this branch will never happen.

Reply:
We are pulling out `foundListAdapter` just before we re-assign its value. If the re-assigned `foundListAdapter` is falsey then we can still through a useful error message containing the `key` name of the last good `foundListAdapter`.